pyvista/plotting/axes_assembly.py

```diff
@@ -465,43 +478,212 @@ def label_color(self, color: ColorLike):  # numpydoc ignore=GL08
+
+        Set :attr:`~pyvista.Property.opacity` for the x-axis only. The property is set
+        for both the axis shaft and tip by default.
+
+        >>> axes_actor.set_part_prop('opacity', 0.5, axis='x')
+        >>> axes_actor.get_part_prop('opacity')
+        _AxesPropTuple(x_shaft=0.5, y_shaft=1.0, z_shaft=1.0, x_tip=0.5, y_tip=1.0, z_tip=1.0)
+
+        Set the property again, but this time set separate values for the shaft and tip.
+
+        >>> axes_actor.set_part_prop('opacity', [0.3, 0.7], axis='x')
+        >>> axes_actor.get_part_prop('opacity')
+        _AxesPropTuple(x_shaft=0.3, y_shaft=1.0, z_shaft=1.0, x_tip=0.7, y_tip=1.0, z_tip=1.0)
+
+        Set :attr:`~pyvista.Property.show_edges` for the axes shafts only. The property
+        is set for all axes by default.
+
+        >>> axes_actor.set_part_prop('show_edges', True, part='shaft')
+        >>> axes_actor.get_part_prop('show_edges')
+        _AxesPropTuple(x_shaft=True, y_shaft=True, z_shaft=True, x_tip=False, y_tip=False, z_tip=False)
+
+        Set the property again, but this time set separate values for each shaft.
+
+        >>> axes_actor.set_part_prop(
+        ...     'show_edges', [True, False, True], part='shaft'
+        ... )
+        >>> axes_actor.get_part_prop('show_edges')
+        _AxesPropTuple(x_shaft=True, y_shaft=False, z_shaft=True, x_tip=False, y_tip=False, z_tip=False)
+
+        Set :attr:`~pyvista.Property.style` for a single axis and specific part.
+
+        >>> axes_actor.set_part_prop(
+        ...     'style', 'wireframe', axis='x', part='shaft'
+        ... )
+        >>> axes_actor.get_part_prop('style')
+        _AxesPropTuple(x_shaft='Wireframe', y_shaft='Surface', z_shaft='Surface', x_tip='Surface', y_tip='Surface', z_tip='Surface')
+        """
+        actors = self._filter_part_actors(axis=axis, part=part)
+        values: Sequence[float | str | ColorLike]
+
+        # Validate input as a sequence of values
+        if 'color' in name:
+            # Special case for color inputs
+            if axis == 'all' and part == 'all':
+                n_values = 6
+            elif part == 'all':
+                n_values = 2
+            elif axis == 'all':
+                n_values = 3
+            else:
+                n_values = 1
+            values = _validate_color_sequence(value, n_values)
+        elif isinstance(value, Sequence) and not isinstance(value, str):
+            # Number sequence
+            values = value
+        else:
+            # Scalar number or string
+            values = [value] * len(actors)
+
+        if len(values) != len(actors):
+            raise ValueError(
+                f"Number of values ({len(values)}) in {value} must match the number of parts ({len(actors)}) for axis '{axis}' and part '{part}'"
+            )
+
+        # Sequence is valid, now set values
+        for actor, val in zip(actors, values):
+            setattr(actor.prop, name, val)
+
+    def get_part_prop(self, name: str):
```

Comment:
I've toyed with adding an `'axis'` and `'part'` parameter to the getter too. This would make this use case a bit nicer:
```  diff
- self.get_part_prop('color')[_AxisEnum.z.value :: 3]
+ self.get_part_prop('color', axis='z')

 But then the output could be a scalar, or a length 2, 3, or 6 tuple, which could result in index errors if user's aren't careful with how they use the arguments/outputs. This could maybe be addressed by overloading the typing to hint the output type for all cases...

Should we add these parameters, or keep the status quo and always return a 6-tuple?

Reply:
When specifying parameters, I think the user will be aware of the size, so I don't think usability will be affected even if we change the size.
We are looking forward to your new PR.